src/main/groovy/life/qbic/datamodel/dtos/business/QuotationId.groovy

```diff
@@ -0,0 +1,27 @@
+package life.qbic.datamodel.dtos.business
+
+/**
+ * A DTO describing the quotation identifier
+ *
+ * A quotation identifier helps to distinguish between offers. It is a {@link TomatoId}
+ *
+ * @since: 1.12.0
+ *
+ */
+class QuotationId extends TomatoId{
+    /**
+     * A quotation is identified by the type Q
+     */
+    private static final String type = "Q"
+
+
+    QuotationId(String projectConservedPart, String randomPart, int version) {
+        super(type,projectConservedPart,randomPart,version)
+    }
+
+    @Override
+    String getType() {
+        return type
+    }
+
```

Comment:
```suggestion
```

Reply:
If I remove the getter Id cannot retrieve the type from the extending classes

Reply:
How should my abstract class know about the type variable in the extending classes?

Reply:
The problem here is that the static variable shadows the class field of the abstract class.
Suggestion: remove the static variable and just call the constructor with "Q"

Reply:
> How should my abstract class know about the type variable in the extending classes?

I fear I do not understand your question. How is this field different from all the other fields?